tests/components/ActionButton/ActionButton.spec.ts

```diff
@@ -0,0 +1,120 @@
+        appendTitleSpy = sandbox.spy(<any>ActionButton.prototype, 'appendTitle');
+        appendTooltipSpy = sandbox.spy(<any>ActionButton.prototype, 'appendTooltip');
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    describe('with click option', () => {
+        let clickHandlerSpy: SinonSpy;
+
+        beforeEach(() => {
+            clickHandlerSpy = spy();
+            options.click = clickHandlerSpy;
+            testSubject = createActionButton(options);
+        });
+
+        it('should call the handler when clicking the button', () => {
+            testSubject.element.dispatchEvent(new Event('click', {}));
+
+            expect(clickHandlerSpy.called).toBeTrue();
+        });
+    });
+
+    describe('without icon and title', () => {
+        let renderSpy: SinonSpy;
+
+        beforeEach(() => {
+            renderSpy = sandbox.spy(<any>ActionButton.prototype, 'render');
+            options.title = '';
+            options.icon = '';
+
+            testSubject = createActionButton(options);
+        });
+
+        it('should not render the button', () => {
+            expect(renderSpy.called).toBeFalse();
+        });
+    });
+
+    [
+        {
+            optionName: 'title',
+            optionValue: 'some title',
+            getSpy: () => appendTitleSpy
+        },
+        {
+            optionName: 'icon',
+            optionValue: icons.copy,
+            getSpy: () => appendIconSpy
+        },
+        {
+            optionName: 'tooltip',
+            optionValue: 'Some Tooltip',
+            getSpy: () => appendTooltipSpy
+        }
+    ].forEach(testCase => {
+        describe(`with empty ${testCase.optionName} option`, () => {
+            beforeEach(() => {
+                setOption(testCase.optionName, null);
+                testSubject = createActionButton(options);
+            });
+
+            it(`should not include the ${testCase.optionName}`, () => {
+                expect(testCase.getSpy().called).toBeFalse();
+            });
+        });
+
+        describe(`with ${testCase.optionName} option`, () => {
+            beforeEach(() => {
+                setOption(testCase.optionName, testCase.optionValue);
+                testSubject = createActionButton(options);
+            });
+
+            it(`should include the ${testCase.optionName}`, () => {
+                expect(testCase.getSpy().called).toBeTrue();
+            });
+        });
+    });
+
+    const createActionButton = (options: IActionButtonOptions) => {
+        const element = document.createElement('button');
+        const componentSetup = Mock.advancedComponentSetup<ActionButton>(ActionButton, new Mock.AdvancedComponentSetupOptions(element, options));
+
+        return componentSetup.cmp;
+    };
+
+    const setOption = (optionName: string, optionValue: any) => {
+        const dictOptions = options as { [key: string]: any };
```

Comment:
I think you should define the type on declaration, L20

Reply:
I don't think so. The purpose of this method is solely to be able to dynamically assign option values for testing. The type `{ [key: string]: any }` is expected to be used only in the scope of this particular method.